sympy/core/tests/test_sympify.py

```diff
@@ -769,3 +769,8 @@ def test_issue_14706():
     raises(SympifyError, lambda: sympify(numpy.array([1]), strict=True))
     raises(SympifyError, lambda: sympify(z1, strict=True))
     raises(SympifyError, lambda: sympify(z2, strict=True))
+
+def test_sympify_expected_type():
+    assert _sympify(5, Integer) == Integer(5)
+    assert _sympify(1.7, Float) == Float(1.7)
+    raises(NotImplementedError, lambda: _sympify(1.7, float))
```

Comment:
It would make more sense to test generic types here like `Set`, `Expr`, `Boolean`, `Number`.